save-demo/src/main/kotlin/com/saveourtool/save/demo/runners/cli/DiktatCliRunner.kt

```diff
@@ -32,57 +32,41 @@ import kotlin.io.path.*
 @Component
 class DiktatCliRunner(
     private val toolStorage: ToolStorage,
-) : CliRunner<DiktatDemoAdditionalParams, DiktatDemoResult> {
-    override fun getExecutable(workingDir: Path, params: DiktatDemoAdditionalParams): Path {
-        val toolName = params.tool.name.lowercase()
-        val version = if (params.tool == DiktatDemoTool.DIKTAT) {
-            DIKTAT_VERSION
-        } else {
-            KTLINT_VERSION
+) : CliRunner<DiktatDemoAdditionalParams, ToolKey, DiktatDemoResult> {
+    private fun getRunCommandForDiktat(
+        workingDir: Path,
+        testPath: Path,
+        outputPath: Path,
+        configPath: Path?,
+        params: DiktatDemoAdditionalParams,
+    ): String = buildString {
+        // TODO: this information should not be hardcoded but stored in database
+        val ktlintExecutable = getExecutable(workingDir, DiktatDemoTool.KTLINT.toToolKey("ktlint"))
+        val diktatExecutable = getExecutable(workingDir, DiktatDemoTool.DIKTAT.toToolKey("diktat-1.2.3.jar"))
+        append(ktlintExecutable)
+        append(" -R $diktatExecutable ")
+        append(" --disabled_rules=standard")
+        append(" --reporter=plain,output=$outputPath ")
+        configPath?.let {
+            append(" --config=$it ")
         }
-        val fileName = buildString {
-            append(toolName)
-            if (params.tool == DiktatDemoTool.KTLINT) {
-                append("-cli")
-            }
-            if (isWindows()) {
-                append(".cmd")
-            }
+        if (params.mode == DiktatDemoMode.FIX) {
+            append(" --format ")
         }
-        val key = ToolKey(toolName, version, fileName)
-
-        return Mono.zip(
-            key.toMono(),
-            toolStorage.doesExist(key)
-        )
-            .filter { (_, doesExist) ->
-                doesExist
-            }
-            .switchIfEmpty {
-                throw FileNotFoundException("Could not find file with key $key")
-            }
-            .flatMapMany { (key, _) ->
-                toolStorage.download(key)
-            }
-            .collectToFile(workingDir / key.executableName)
-            .thenReturn(workingDir / key.executableName)
-            .block()
-            .let { requireNotNull(it) }
-            .apply {
-                toFile().setExecutable(true, false)
-            }
+        append(testPath)
     }
 
-    override fun getRunCommand(
+    private fun getRunCommandForKtlint(
```

Comment:
Diktat command line is actually a ktlint command line with three extra parameters:

 - `-R`
 - `diktat.jar`
 - `--disabled_rules=standard`
 
 Please refactor so that common code is shared.

Reply:
Done, combined two methods into one with `if (tool == DIKTAT)`